### PR TITLE
Decouple markdown loader

### LINKED ISSUE #83: Decouple Markdown translation logic from Webpack plumbing in mdLoader

### PR DESCRIPTION
**Pull request by phildarnowsky-broad**
resolves #83

### CODE REVIEW

**Comment by phildarnowsky-broad** [comment]
@mattsolo1 @rileyhgrant please see https://github.com/broadinstitute/gnomad-browser-toolkit/pull/84/files#diff-588ca5e5cd7af3662aa179ed582b7497386471a4fff2905608dc1a0c00fe5020 for an example of how to use the now-exposed Markdown translation. In particular @rileyhgrant note that you'll get back a promise rather than a proper value, so to write your browser snapshot test we were discussing using this, you'll need to read https://jestjs.io/docs/asynchronous.

File: packages/markdown-loader/example/MarkdownLoaderExample.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
-document.body.innerHTML = example.html
+const { translateMarkdown } = require('../src/markdownLoader')
+
+const literalSource = `
+# Begin first section
+This section was generated via \`translateMarkdown\`. The section below was \`import\`ed from a static Markdown source file.
+# End first section. Second section follows.
+`
+
+translateMarkdown(literalSource).then(markdown => {
```
Comment (mattsolo1) [inline packages/markdown-loader/example/MarkdownLoaderExample.js:13]: Just to be clear... these changes are backwards compatible and it's optional to use `translateMarkdown`? 

Hard to tell since the example now only uses the later.

File: packages/markdown-loader/example/ExampleContent.md [hunk 2]
```diff
@@ -1,7 +1,313 @@
 ---
 title: Markdown example
 ---
+# Markdown: Syntax
 
-# Hello World
+*   [Overview](#overview)
```
Comment (mattsolo1) [inline packages/markdown-loader/example/ExampleContent.md:6]: I like that you've expanded the example

File: packages/markdown-loader/example/MarkdownLoaderExample.js (hunk 1 above)
Comment (phildarnowsky-broad) [reply packages/markdown-loader/example/MarkdownLoaderExample.js:13]: it's perhaps hard to see in the visual noise of the diff but we are still also using the established `import foo.md` method in the example, notice in this line:

```
document.body.innerHTML = [markdown.contents, importedExample.html].join(' ')
```

That's just a simple (or maybe simple-minded) way of testing both ways at once: `importedExample` has the content we `import`ed from a file, and `markdown` (which name I have to change because it's entirely misleading) has the content we generated via calling `translateMarkdown`. So if all goes well, we should get HTML consisting of the rendered-to-HTML literal content, followed by the rendered-to-HTML imported content.

To recap, backwards compatibility is as you say key here: we didn't change the existing public API or its behavior at all. `markdownLoader` still exists and still works the same way, but all the actual work of translating Markdown to HTML is now offloaded to `translateMarkdown`, leaving `markdownLoader` as just glue code that exposes that functionality to Webpack.

**Review by mattsolo1 — APPROVED** [review]

**Comment by phildarnowsky-broad** [comment]
@mattsolo1 @rileyhgrant note that, while I had meant for this to have a test that the modified code actually works with Webpack, I've just spent a couple of days on that without getting anywhere near a test that'll run--in theory, it should be possible to create a Webpack instance with a special configuration for the sole purpose of exercising this loader in a test, but in practice, apparently I don't know how.

To have at least some assurance that this didn't break the build, I tested locally by hand, ensuring that, when hooked up to `gnomad-browser-toolkit` with this changes incorporated, I could still `yarn build` `gnomad-browser`, and that the help pages on my dev server still work the same as always.

There are other choices of bundler these days, which are said to have a lot of advantages over Webpack, but I don't know if that's actually a worthwhile idea or just the frustration talking. I sort of doubt whether this would be an advantageous time to make a switch, but if you're interested I could look more into that.

**Comment by phildarnowsky-broad** [comment]
Going to override the branch protections, there are still failing tests but not in a package I touched, and in any case there are fewer overall failures than before.

**Comment by phildarnowsky-broad** [comment]
@rileyhgrant it's late in the day to start now, but I plan to start tomorrow by publishing this to NPM, so after that you should be able to return to those help-page tests in `gnomad-browser` if you want.

**Comment by rileyhgrant** [comment]
> @rileyhgrant it's late in the day to start now, but I plan to start tomorrow by publishing this to NPM, so after that you should be able to return to those help-page tests in `gnomad-browser` if you want.

Shweet! Thanks for keeping me in the loop on this - excited to get snapshot test for the more static pages up and running

**Comment by phildarnowsky-broad** [comment]
@rileyhgrant 

> Shweet! Thanks for keeping me in the loop on this - excited to get snapshot test for the more static pages up and running

Yeah, I think we might like to revisit the build process for those pages in the future, and I would feel so much better about doing so with a safety net in place.
